app/helpers/post_helper.rb

```diff
@@ -13,7 +13,9 @@ def author_links(post, linked: true, colored: false)
       return safe_join([joined_links, "#{num_deleted} #{deleted}"], ' and ')
     end
 
-    first_author = post.user.deleted? ? authors.first : post.user
+    # filter post.user from post.authors to avoid n+1 query over post.user
+    first_author = authors.find { |x| x.id == post.user_id } || authors.first
```

Comment:
Any posts with "<user> and 3 others" currently use an n+1 to select the first post author

Reply:
you lost the deletion check there
```suggestion
    first_author = authors.active.find { |x| x.id == post.user_id } || authors.first
```

Reply:
`authors` is defined above and already rejects deleted, actually!

```
authors = post.authors.reject(&:deleted?).sort_by { |a| a.username.downcase }
```

Reply:
oh, excellent, carry on then. -- possibly we should check if `post.user_deleted` exists before we do the more computationally expensive thing but it's not a big deal